File: apps/consumer/android/wear/src/main/java/com/orderingintelligence/consumer/wear/WatchModels.kt
package com.orderingintelligence.consumer.wear

data class WatchReorderItem(val itemId: String, val quantity: Int)

data class WatchSessionInfo(
  val storeId: String,
  val storeName: String,
  val businessType: String,
  val currency: String,
  val fuelDefaultPrepayCents: Int,
  val fuelPreauthCapCents: Int
)

data class WatchFuelOrder(
  val fuelGradeId: String,
  val fuelGradeName: String,
  val unitPriceCents: Int,
  val requestedLiters: Double,
  val requestedAmountCents: Int,
  val preauthAmountCents: Int,
  val paymentFlow: String
)

data class WatchReorder(
  val storeId: String,
  val storeName: String,
  val title: String,
  val items: List<WatchReorderItem>,
  val currency: String,
  val fuel: WatchFuelOrder? = null
)

data class WatchPaymentMethod(
  val id: String,
  val isDefault: Boolean
)

data class WatchOffSessionResult(
  val status: String,
  val paymentId: String?,
  val paymentIntentId: String?,
  val clientSecret: String?,
  val customerId: String?,
  val ephemeralKey: String?,
  val stripeAccountId: String?,
  val publishableKey: String?,
  val amountCents: Int?,
  val currency: String?,
  val error: String?
) {
  val requiresAction: Boolean
    get() = status == "requires_action"
  val succeeded: Boolean
    get() = status == "succeeded" || status == "requires_capture" || status == "processing"
}

data class WatchOrderPaymentResult(
  val orderId: String,
  val payment: WatchOffSessionResult
)

File: apps/consumer/ios/WatchApp/WatchModels.swift
import Foundation

struct WatchReorderItem: Decodable {
  let itemId: String
  let quantity: Int
}

struct WatchSessionInfo: Decodable {
  let storeId: String?
  let storeName: String?
  let businessType: String?
  let currency: String?
  let fuelDefaultPrepayCents: Int?
  let fuelPreauthCapCents: Int?
}

struct WatchFuelOrder: Decodable {
  let fuelGradeId: String
  let fuelGradeName: String?
  let unitPriceCents: Int?
  let requestedLiters: Double?
  let requestedAmountCents: Int?
  let preauthAmountCents: Int?
  let paymentFlow: String?
}

struct WatchReorder: Decodable, Identifiable {
  let storeId: String
  let storeName: String
  let title: String
  let items: [WatchReorderItem]
  let fuel: WatchFuelOrder?
  let currency: String?

  var id: String { "\(storeId)_\(title)" }
}

struct WatchReorderResponse: Decodable {
  let results: [WatchReorder]
}

struct WatchPaymentMethod: Decodable {
  let id: String
  let isDefault: Bool
}

struct WatchPaymentMethodsResponse: Decodable {
  let methods: [WatchPaymentMethod]
}

struct WatchOffSessionResult: Decodable {
  let status: String
  let paymentId: String?
  let paymentIntentId: String?
  let clientSecret: String?
  let customerId: String?
  let ephemeralKey: String?
  let stripeAccountId: String?
  let publishableKey: String?
  let amountCents: Int?
  let currency: String?
  let error: String?

  var requiresAction: Bool { status == "requires_action" }
  var succeeded: Bool {
    return status == "succeeded" || status == "requires_capture" || status == "processing"
  }
}

struct WatchOrderPaymentResult {
  let orderId: String
  let payment: WatchOffSessionResult
}
